File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const map=new Map()
    
    for(let name of participant){
        if (map.has(name)) {
            map.set(name, map.get(name) + 1);
        } else {
            map.set(name, 1);
        }
    }
    
    for(let name of completion){
        map.set(name, map.get(name) - 1);
    }
    
    for(const [key, value] of map){
        if(value>0) return key
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<>();
        
        for(String name : participant){
            map.put(name, map.getOrDefault(name,0)+1);
        }
        
        for(String name : completion){
            if(map.get(name)>1) map.put(name, map.get(name)-1);
            else map.remove(name);
        }
        for (String key : map.keySet()){
            return key;
        }
        return map.keySet().iterator().next();
    }
}